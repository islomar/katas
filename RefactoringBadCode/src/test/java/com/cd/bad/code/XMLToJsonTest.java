package com.cd.bad.code;

import org.approvaltests.Approvals;
import org.junit.Test;

import java.net.URL;

import static org.junit.Assert.assertEquals;

public class XMLToJsonTest
{
    @Test
    public void shouldTranslateEmptyXMLToJson() throws Exception
    {
        XMLToJson translate = new XMLToJson();

        URL url = new URL("file:./src/test/resources/toc.xml");
        String xPathString = "fk:AMM24_fk:AMM24-00-00_fk:AMM24-00-00-02";

        Approvals.verify(translate.getJson(url, xPathString));
    }
}
