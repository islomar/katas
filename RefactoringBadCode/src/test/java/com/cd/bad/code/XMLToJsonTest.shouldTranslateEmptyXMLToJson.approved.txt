[{'data':'TASK 24-00-00-700-801 - AC Power, DC Power and Battery Maintenance Practice Recommendations','attr':{'id':'fk:AMM24_fk:AMM24-00-00_fk:AMM24-00-00-02_dk:AMM24-00-00-700-801','file':'AMM24-00-00-700-801.pdf'}},{'data':'TASK 24-00-00-700-801 - AC Power, DC Power and Battery Maintenance Practice Recommendations','attr':{'id':'fk:AMM24_fk:AMM24-00-00_fk:AMM24-00-00-02_dtrn:1234','file':'AMM24-00-00-700-801.pdf'}},{'data':'List of Effective TRs','attr':{'id':'fk:AMM24_fk:AMM24-00-00_fk:AMM24-00-00-02_fk:LOETR'}','file':'loetr.html'}},{'data':'List of Effective TRs',},{'data':'History of AMM12-31-00-660-806',}]